workflow/util/util_test.go

```diff
@@ -1388,14 +1374,13 @@ func TestFormulateRetryWorkflow(t *testing.T) {
 			},
 		}
 		_, err := wfClient.Create(ctx, wf, metav1.CreateOptions{})
-		assert.NoError(t, err)
+		require.NoError(t, err)
 		wf, podsToDelete, err := FormulateRetryWorkflow(ctx, wf, true, "id=3", nil)
-		if assert.NoError(t, err) {
-			if assert.Len(t, wf.Status.Nodes, 2) {
-				assert.Equal(t, wfv1.NodeSucceeded, wf.Status.Nodes["1"].Phase)
-				assert.Equal(t, wfv1.NodeRunning, wf.Status.Nodes["continue-on-failed-workflow-2"].Phase)
-				assert.Len(t, podsToDelete, 4)
-			}
+		require.NoError(t, err)
+		if assert.Len(t, wf.Status.Nodes, 2) {
+			assert.Equal(t, wfv1.NodeSucceeded, wf.Status.Nodes["1"].Phase)
+			assert.Equal(t, wfv1.NodeRunning, wf.Status.Nodes["continue-on-failed-workflow-2"].Phase)
+			assert.Len(t, podsToDelete, 4)
 		}
```

Comment:
```suggestion
		require.Len(t, wf.Status.Nodes, 2)
		assert.Equal(t, wfv1.NodeSucceeded, wf.Status.Nodes["1"].Phase)
		assert.Equal(t, wfv1.NodeRunning, wf.Status.Nodes["continue-on-failed-workflow-2"].Phase)
		assert.Len(t, podsToDelete, 4)
```

this can also be simplified as a `require` if I'm not mistaken

Reply:
see also my comment on the other PR: https://github.com/argoproj/argo-workflows/pull/13401#discussion_r1696246571

these are not `assert.Error` changes though, so may make sense as a separate PR. as it would be another indentation change though, it may make sense to consolidate for diff simplicity